lib/repl.js

```diff
@@ -386,14 +386,15 @@ function REPLServer(prompt,
   self.bufferedCommand = '';
   self.lines.level = [];
 
-  function complete(text, callback) {
-    self.complete(text, callback);
-  }
+  // Figure out which "complete" function to use.
+  self.completer = (typeof options.completer === 'function')
+    ? options.completer
+    : complete.bind(self);
```

Comment:
Have you tested and seen this behavior? Prior to this change, `complete()` was not bound. You also don't seem to need to bind any of the custom completers in your tests.